### PR TITLE
[4059] Fix form engine close exception

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4059

### CODE REVIEW

**Comment by jvega190** [comment]
Pending unlock on unload event, need to discuss

File: ui/app/src/services/content.ts [hunk 1]
```diff
@@ -1236,5 +1236,6 @@ export default {
   fetchQuickCreateList,
   fetchLegacyContentTypes,
   getContentHistory: getHistory,
-  revertTo
+  revertTo,
+  unlock
```
Comment (rart) [inline ui/app/src/services/content.ts]: This is outdated. Pull.

File: static-assets/components/cstudio-forms/forms-engine.js [hunk 2]
```diff
@@ -1708,20 +1710,25 @@ var CStudioForms =
...
-                  if (componentsOn) {
-                    CStudioAuthoring.Operations.refreshPreviewParent();
-                  }
-                } else {
-                  window.close();
-                  if (componentsOn) {
-                    CStudioAuthoring.Operations.refreshPreviewParent();
-                  }
+
+                  YAHOO.util.Event.removeListener(window, 'unload', unloadFn, me);
```
Comment (rart) [inline static-assets/components/cstudio-forms/forms-engine.js]: Unload should be removed once the unlock has succeeded.

File: static-assets/components/cstudio-forms/forms-engine.js [hunk 3]
```diff
@@ -1388,6 +1388,8 @@ var CStudioForms =
                 edit
               });
             } else {
+              YAHOO.util.Event.removeListener(window, 'unload', unloadFn, me);
```
Comment (rart) [inline static-assets/components/cstudio-forms/forms-engine.js]: Unload should be removed once the unlock has succeeded.

**Review by rart — CHANGES_REQUESTED** [review]
- See comments.
- Let's fix 3.1.x too please.

File: static-assets/components/cstudio-forms/forms-engine.js [hunk 4]
```diff
@@ -1562,11 +1564,12 @@ var CStudioForms =
           };
 
           var unloadFn = function(e) {
+            console.log('UNLOAD');
```
Comment (rart) [inline static-assets/components/cstudio-forms/forms-engine.js]: x

**Review by rart — APPROVED** [review]

**Pull request merged**